File: Export/Natives/Name/FileNamePartProvider.cs
using kCura.WinEDDS.Core.Model;
using kCura.WinEDDS.Exporters;

namespace kCura.WinEDDS.Core.Export.Natives.Name
{
	
	public abstract class FileNamePartProvider<T> : IFileNamePartProvider<T> where T : DescriptorPart
	{
		public abstract string GetPartName(T descriptorPart, ObjectExportInfo exportObject);

		public virtual string GetPartName(DescriptorPart descriptorDescriptorPartBase, ObjectExportInfo exportObject)
		{
			return GetPartName(descriptorDescriptorPartBase as T, exportObject);
		}
	}
}

File: Source/Relativity.Import.Export.Legacy/FileNaming/CustomFileNaming/FileNamePartProvider.vb
Imports FileNaming.CustomFileNaming
Imports kCura.WinEDDS.Exporters

Namespace kCura.WinEDDS.FileNaming.CustomFileNaming
	Public MustInherit Class FileNamePartProvider(Of T As DescriptorPart)
		Implements IFileNamePartProvider(Of T)

		Public MustOverride Function GetPartName(descriptorPart As T, exportObject As ObjectExportInfo) As String Implements IFileNamePartProvider(Of T).GetPartName

		Public Overridable Function GetPartName(descriptorPart As DescriptorPart, exportObject As ObjectExportInfo) As String Implements IFileNamePartProvider.GetPartName
			Return GetPartName(TryCast(descriptorPart, T), exportObject)
		End Function
	End Class
End Namespace
